pymodm/base/options.py

```diff
@@ -86,8 +86,14 @@ def get_field_from_attname(self, attname):
 
     def add_field(self, field_inst):
         """Add or replace a given Field."""
-        orig_field = (self.get_field(field_inst.mongo_name) or
-                      self.get_field_from_attname(field_inst.attname))
+        try:
+            orig_field = self.get_field(field_inst.mongo_name)
+        except Exception:
+            # FieldDoesNotExist, etc. may be raised by subclasses.
+            try:
+                orig_field = self.get_field_from_attname(field_inst.attname)
```

Comment:
Good question. The answer is yes because a Model can override an existing field and change the 'mongo_name' at the same time, see: https://github.com/mongodb/pymodm/commit/b4aaecdf9d25abc7ff807dafe45ca92a72815c7b#diff-cbb4ddb155212dac74b5eb6898afbb3eR23